File: tests/expected/loop.cpp
#include <cppitertools/range.hpp>  // NOLINT(build/include_order)
#include <cstdint>                 // NOLINT(build/include_order)
#include <iostream>                // NOLINT(build/include_order)
inline void for_with_break() {
  for (auto i : iter::range(4)) {
    if (i == 2) {
      break;
    }
    std::cout << i;
    std::cout << std::endl;
  }
}

inline void for_with_continue() {
  for (auto i : iter::range(4)) {
    if (i == 2) {
      continue;
    }
    std::cout << i;
    std::cout << std::endl;
  }
}

inline void for_with_else() {
  bool has_break = false;
  for (auto i : iter::range(4)) {
    std::cout << i;
    std::cout << std::endl;
  }
  if (has_break != true) {
    std::cout << std::string{"OK"};
    std::cout << std::endl;
  }
}

inline void while_with_break() {
  int i = 0;
  while (true) {
    if (i == 2) {
      break;
    }
    std::cout << i;
    std::cout << std::endl;
    i += 1;
  }
}

inline void while_with_continue() {
  int i = 0;
  while (i < 5) {
    i += 1;
    if (i == 2) {
      continue;
    }
    std::cout << i;
    std::cout << std::endl;
  }
}

int main(int argc, char** argv) {
  for_with_break();
  for_with_continue();
  for_with_else();
  while_with_break();
  while_with_continue();
}

File: tests/expected/loop.kt
fun for_with_break() {
    for (i in (0..4 - 1)) {
        if (i == 2) {
            break
        }
        println("$i")
    }
}

fun for_with_continue() {
    for (i in (0..4 - 1)) {
        if (i == 2) {
            continue
        }
        println("$i")
    }
}

fun for_with_else() {
    val has_break = false
    for (i in (0..4 - 1)) {
        println("$i")
    }
    if (has_break != true) {
        println("OK")
    }
}

fun while_with_break() {
    var i = 0
    while (true) {
        if (i == 2) {
            break
        }
        println("$i")
        i += 1
    }
}

fun while_with_continue() {
    var i = 0
    while (i < 5) {
        i += 1
        if (i == 2) {
            continue
        }
        println("$i")
    }
}

fun main(argv: Array<String>) {
    for_with_break()
    for_with_continue()
    for_with_else()
    while_with_break()
    while_with_continue()
}
